src/operators/infinitepepo.jl

```diff
@@ -153,7 +153,7 @@ domain_physicalspace(T::InfinitePEPO, r::Int, c::Int) = domain_physicalspace(T[r
 function codomain_physicalspace(T::InfinitePEPO, r::Int, c::Int)
     return codomain_physicalspace(T[r, c, end])
 end
-physicalspace(T::InfinitePEPO) = physicalspace.(Ref(T), 1:size(T, 1), 1:size(T, 2))
+physicalspace(T::InfinitePEPO) = physicalspace.(@view(unitcell(T)[:, :, end]))
```

Comment:
I agree that the previous implementation was wrong, although I dont think this is fully right either. Previously we were verifying that the `codomain_physicalspace` of the last layer and the `domain_physicalspace` of the first layer match, now this only verifies the last layer, so if we would have two layers that alternate between `ComplexSpace(2)` and `ComplexSpace(3)`, this would now error.

```suggestion
physicalspace(T::InfinitePEPO) = [physicalspace(T, row, col) for row in axes(T, 1), col in axes(T, 2)]
```